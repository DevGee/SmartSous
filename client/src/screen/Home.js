import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Icon } from 'react-native-elements';
import { LoginManager } from 'react-native-fbsdk';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    paddingTop: 23,
  },
  logoutButton: {
    position: 'absolute',
    right: 0,
    top: 23,
    zIndex: 1,
  },
  barCodeButton: {

  },
  communityFridge: {

  },
  top: {
    flex: 1,
    flexDirection: 'row',
    alignSelf: 'stretch',
  },
  bottom: {
    flex: 1,
    flexDirection: 'row',
    alignSelf: 'stretch',
  },
  quadrantOne: {
    flex: 1,
    alignItems: 'center',
    alignSelf: 'stretch',
    justifyContent: 'center',
    backgroundColor: 'lightblue',
  },
  quadrantTwo: {
    flex: 1,
    alignItems: 'center',
    alignSelf: 'stretch',
    justifyContent: 'center',
    backgroundColor: 'lightskyblue',
  },
  quadrantThree: {
    flex: 1,
    alignItems: 'center',
    alignSelf: 'stretch',
    justifyContent: 'center',
    backgroundColor: 'lightskyblue',
  },
  quadrantFour: {
    flex: 1,
    alignItems: 'center',
    alignSelf: 'stretch',
    justifyContent: 'center',
    backgroundColor: 'lightblue',
  },
  textTitle: {
    paddingTop: 15,
    fontSize: 16,
  },
});

class Home extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.top}>
          <View style={styles.quadrantOne}>
            <Icon
              reverse
              raised
              size={36}
              containerStyle={styles.communityFridge}
              name='ios-people'
              type='ionicon'
              onPress={() => this.props.navigation.navigate('ViewCommScreen')}
            />
            <Text style={styles.textTitle}>View Community</Text>
          </View>
          <View style={styles.quadrantTwo}>
            <Icon
              reverse
              raised
              size={36}
              containerStyle={styles.communityFridge}
              name='plus-square-o'
              type='font-awesome'
              onPress={() => this.props.navigation.navigate('MakeCommScreen')}
            />
            <Text style={styles.textTitle}>Make Community</Text>
          </View>
        </View>
        <View style={styles.bottom}>
          <View style={styles.quadrantThree}>
            <Icon
              reverse
              raised
              size={36}
              containerStyle={styles.communityFridge}
              name='group-add'
              type='material-icons'
              onPress={() => this.props.navigation.navigate('JoinCommScreen')}
            />
            <Text style={styles.textTitle}>Join Community</Text>
          </View>
          <View style={styles.quadrantFour}>
            <Icon
              reverse
              raised
              size={36}
              containerStyle={styles.barCodeButton}
              name='barcode-scan'
              type='material-community'
              onPress={() => this.props.navigation.navigate('BarCodeScreen')}
            />
            <Text style={styles.textTitle}>Scan Barcode</Text>
          </View>
        </View>
      </View>
    );
  }
}

export default Home;
