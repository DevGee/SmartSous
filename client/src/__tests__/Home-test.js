import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import Home from '../screen/Home';

test('renders correctly', () => {
  const tree = renderer.create(<Home/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
