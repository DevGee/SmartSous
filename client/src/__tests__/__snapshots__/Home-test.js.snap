// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "paddingTop": 23,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          Object {
            "margin": 7,
          },
          Object {
            "borderRadius": 19,
            "height": 34,
            "width": 34,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "justifyContent": "center",
          },
          Object {
            "position": "absolute",
            "right": 0,
            "top": 23,
          },
        ],
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 15,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "lightblue",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "margin": 7,
              },
              Object {
                "borderRadius": 40,
                "height": 76,
                "width": 76,
              },
              Object {
                "shadowColor": "rgba(0,0,0, .4)",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 1,
                "shadowRadius": 1,
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "justifyContent": "center",
              },
              Object {},
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 36,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "red",
          "flex": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "green",
          "flex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "yellow",
          "flex": 1,
        }
      }
    />
  </View>
</View>
`;
